e2etest/l2tests/assignment.go

```diff
@@ -145,6 +148,63 @@ var _ = ginkgo.Describe("IP Assignment", func() {
 					err := cs.CoreV1().Services(svc.Namespace).Delete(context.Background(), svc.Name, metav1.DeleteOptions{})
 					return err
 				}))
+
+		ginkgo.It("should preseve the same external ip after controller restart", func() {
+			const numOfRestarts = 5
+			resources := config.Resources{
+				Pools: []metallbv1beta1.IPAddressPool{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "assignment-controller-reset-test-pool",
+						},
+						Spec: metallbv1beta1.IPAddressPoolSpec{
+							Addresses: []string{"192.168.10.100/32"},
+						},
+					},
+				},
+			}
+			err := ConfigUpdater.Update(resources)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("creating 4 LB services")
+			jig := e2eservice.NewTestJig(cs, f.Namespace.Name, "service-specific")
+			specificService, err := jig.CreateLoadBalancerService(30*time.Second, func(svc *v1.Service) {})
+			framework.ExpectNoError(err)
+			defer service.Delete(cs, specificService)
+			service.ValidateDesiredLB(specificService)
+
+			svc, err := jig.CreateLoadBalancerServiceWaitForClusterIPOnly(func(svc *v1.Service) { svc.ObjectMeta.Name = "service-b" })
+			framework.ExpectNoError(err)
+			defer service.Delete(cs, svc)
+
+			svc, err = jig.CreateLoadBalancerServiceWaitForClusterIPOnly(func(svc *v1.Service) { svc.ObjectMeta.Name = "service-c" })
```

Comment:
if you override svc, then we'll delete only one service at the end

Reply:
fixed thanks. Notice that I must create new Jig object for each service.

Reply:
fixed